apps/etl-func/src/domain/message-status.ts

```diff
@@ -23,27 +23,48 @@ const statusEnum = z.enum([
   "REJECTED",
 ]);
 
-const rejectedMessageStatusSchema = z.object({
-  fiscalCode: fiscalCodeSchema,
-  isArchived: z.boolean().default(false),
-  isRead: z.boolean().default(false),
-  messageId: z.string().ulid(),
-  rejection_reason: rejectionReasonSchema,
-  status: statusEnum.extract(["REJECTED"]),
-  updatedAt: z.number(),
-});
-
-const notRejectedMessageStatusSchema = z.object({
-  fiscalCode: fiscalCodeSchema,
-  isArchived: z.boolean().default(false),
-  isRead: z.boolean().default(false),
-  messageId: z.string().ulid(),
-  status: statusEnum.exclude(["REJECTED"]),
-  updatedAt: z.number(),
-});
+export const messageStatusSchema = z
+  .object({
+    fiscalCode: fiscalCodeSchema,
+    id: z.string().min(1),
+    isArchived: z.boolean().default(false),
+    isRead: z.boolean().default(false),
+    messageId: z.string().ulid(),
+    updatedAt: z.number(),
+    version: z.number().gte(0),
+  })
+  .and(
+    z.discriminatedUnion("status", [
+      z.object({
+        status: statusEnum.exclude(["REJECTED"]),
+      }),
+      z.object({
+        rejection_reason: rejectionReasonSchema,
+        status: statusEnum.extract(["REJECTED"]),
+      }),
+    ]),
+  );
+export type MessageStatus = z.TypeOf<typeof messageStatusSchema>;
 
-export const messageStatusSchema = z.discriminatedUnion("status", [
-  notRejectedMessageStatusSchema,
-  rejectedMessageStatusSchema,
+export const messageStatusEventOperationSchema = z.enum([
+  "CREATE",
+  "UPDATE",
+  "DELETE",
 ]);
-export type MessageStatus = z.TypeOf<typeof messageStatusSchema>;
+export type MessageStatusEventOperation = z.TypeOf<
+  typeof messageStatusEventOperationSchema
+>;
+
+export const messageStatusEventSchema = z.object({
+  created_at: z.number(),
+  id: z.string().min(1),
```

Comment:
when possible you should try to extract schemas from existing definition - having two schema definition for `messageStatusId` can be a problem in the long run